diff --git a/org/apache/aries/spifly/BaseActivator.java b/org/apache/aries/spifly/BaseActivator.java
index ad34f13cda664..5098491e0a7eb 100644
--- a/org/apache/aries/spifly/BaseActivator.java
+++ b/org/apache/aries/spifly/BaseActivator.java
@@ -171,7 +171,7 @@ public abstract class BaseActivator implements BundleActivator {
 
         List<String> l = new ArrayList<String>();
         for (Bundle bf : bundlesFragments) {
-            String header = bf.getHeaders().get(headerName);
+            String header = bf.getHeaders("").get(headerName);
             if (header != null) {
                 l.add(header);
             }
@@ -394,3 +394,4 @@ public abstract class BaseActivator implements BundleActivator {
     }
 
 }
+/* @generated */
\ No newline at end of file
diff --git a/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java b/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java
index 3f1218f5f9286..328a9a4289c78 100644
--- a/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java
+++ b/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java
@@ -88,7 +88,7 @@ public class ProviderBundleTrackerCustomizer implements BundleTrackerCustomizer
         DiscoveryMode discoveryMode = DiscoveryMode.SERVICELOADER_CAPABILITIES;
         List<String> providedServices = null;
         Map<String, Object> customAttributes = new HashMap<String, Object>();
-        if (bundle.getHeaders().get(SpiFlyConstants.REQUIRE_CAPABILITY) != null) {
+        if (bundle.getHeaders("").get(SpiFlyConstants.REQUIRE_CAPABILITY) != null) {
             try {
                 providedServices = readServiceLoaderMediatorCapabilityMetadata(bundle, customAttributes);
             } catch (InvalidSyntaxException e) {
@@ -275,7 +275,7 @@ public class ProviderBundleTrackerCustomizer implements BundleTrackerCustomizer
             serviceFileURLs.addAll(Collections.list(entries));
         }
 
-        Object bcp = bundle.getHeaders().get(Constants.BUNDLE_CLASSPATH);
+        Object bcp = bundle.getHeaders("").get(Constants.BUNDLE_CLASSPATH);
         if (bcp instanceof String) {
             for (String entry : ((String) bcp).split(",")) {
                 entry = entry.trim();
@@ -297,7 +297,7 @@ public class ProviderBundleTrackerCustomizer implements BundleTrackerCustomizer
     }
 
     private String getHeaderFromBundleOrFragment(Bundle bundle, String headerName, String matchString) {
-        Parameters headerParameters = new Parameters(bundle.getHeaders().get(headerName));
+        Parameters headerParameters = new Parameters(bundle.getHeaders("").get(headerName));
         if (matches(headerParameters.toString(), matchString) && !MERGE_HEADERS.contains(headerName)) {
             return headerParameters.isEmpty() ? null : headerParameters.toString();
         }
@@ -308,7 +308,7 @@ public class ProviderBundleTrackerCustomizer implements BundleTrackerCustomizer
             if (wiring != null) {
                 for (BundleWire wire : wiring.getProvidedWires("osgi.wiring.host")) {
                     Bundle fragment = wire.getRequirement().getRevision().getBundle();
-                    Parameters fragmentParameters = new Parameters(fragment.getHeaders().get(headerName));
+                    Parameters fragmentParameters = new Parameters(fragment.getHeaders("").get(headerName));
                     if (MERGE_HEADERS.contains(headerName)) {
                         headerParameters.mergeWith(fragmentParameters, false);
                     }
@@ -506,3 +506,4 @@ public class ProviderBundleTrackerCustomizer implements BundleTrackerCustomizer
         private final String serviceType;
     }
 }
+/* @generated */
\ No newline at end of file
diff --git a/org/apache/aries/spifly/Util.java b/org/apache/aries/spifly/Util.java
index fab251b1a9959..076ddb0bf8074 100644
--- a/org/apache/aries/spifly/Util.java
+++ b/org/apache/aries/spifly/Util.java
@@ -274,7 +274,7 @@ public class Util {
         }
 
         // if we can't find any classes in the bundle directly, try the Bundle-ClassPath
-        Object bcp = b.getHeaders().get(Constants.BUNDLE_CLASSPATH);
+        Object bcp = b.getHeaders("").get(Constants.BUNDLE_CLASSPATH);
         if (bcp instanceof String) {
             for (String entry : ((String) bcp).split(",")) {
                 entry = entry.trim();
@@ -391,3 +391,4 @@ public class Util {
         }
     }
 }
+/* @generated */
\ No newline at end of file
