diff --git a/org/apache/jasper/JspCompilationContext.java b/org/apache/jasper/JspCompilationContext.java
index d8f926cb4cabb..5c47884dc0448 100644
--- a/org/apache/jasper/JspCompilationContext.java
+++ b/org/apache/jasper/JspCompilationContext.java
@@ -21,6 +21,7 @@ import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.net.JarURLConnection;
 import java.net.MalformedURLException;
+import java.net.URI;
 import java.net.URL;
 import java.net.URLClassLoader;
 import java.net.URLConnection;
@@ -177,6 +178,24 @@ public class JspCompilationContext {
     }
 
     public ClassLoader getJspLoader() {
+        if (baseUrl == null) {
+            File file = options.getScratchDir();
+
+            URI uri = file.toURI();
+            String baseUri = uri.toString();
+            if (!baseUri.endsWith("/")) {
+                baseUri = baseUri.concat("/");
+            }
+
+            try {
+                baseUrl = new URL(baseUri);
+            }
+            catch (MalformedURLException murle) {
+                throw new IllegalArgumentException(
+                    "Unable to use " + file, murle);
+            }
+        }
+
         if( jspLoader == null ) {
             jspLoader = new JasperLoader(new URL[] {baseUrl}, getClassLoader(),
                     rctxt.getPermissionCollection());
@@ -688,7 +707,12 @@ public class JspCompilationContext {
         // Append servlet or tag handler path to scratch dir
         try {
             File base = options.getScratchDir();
-            baseUrl = base.toURI().toURL();
+            URI uri = base.toURI();
+            String baseUri = uri.toString();
+            if (!baseUri.endsWith("/")) {
+                baseUri = baseUri.concat("/");
+            }
+            baseUrl = new URL(baseUri);
             outputDir = base.getAbsolutePath() + File.separator + path + File.separator;
             if (!makeOutputDir()) {
                 log.error(Localizer.getMessage("jsp.error.outputfolder.detail", outputDir));
@@ -767,3 +791,4 @@ public class JspCompilationContext {
         return result.toString();
     }
 }
+/* @generated */
\ No newline at end of file
