diff --git a/com/sun/faces/application/applicationimpl/Version.java b/com/sun/faces/application/applicationimpl/Version.java
index 094497c0e39b2..354765c6dfb33 100644
--- a/com/sun/faces/application/applicationimpl/Version.java
+++ b/com/sun/faces/application/applicationimpl/Version.java
@@ -58,7 +58,7 @@ public class Version {
             
             if (beanManager == null) {
                 // TODO: use version enum and >=
-                if (getFacesConfigXmlVersion(facesContext).equals("2.3") || getWebXmlVersion(facesContext).equals("4.0")) {
+                if (getWebXmlVersion(facesContext).equals("4.0")) {
                     throw new FacesException("Unable to find CDI BeanManager");
                 }
                 isJsf23 = false;
@@ -125,3 +125,4 @@ public class Version {
     }
 
 }
+/* @generated */
\ No newline at end of file
diff --git a/com/sun/faces/el/ELUtils.java b/com/sun/faces/el/ELUtils.java
index 17468e5f1d7d6..56f55f3090782 100644
--- a/com/sun/faces/el/ELUtils.java
+++ b/com/sun/faces/el/ELUtils.java
@@ -332,7 +332,7 @@ public class ELUtils {
         
         if (beanManager == null) {
             // TODO: use version enum and >=
-            if (getFacesConfigXmlVersion(facesContext).equals("2.3") || getWebXmlVersion(facesContext).equals("4.0")) {
+            if (getWebXmlVersion(facesContext).equals("4.0")) {
                 throw new FacesException("Unable to find CDI BeanManager");
             }
         } else {
@@ -965,3 +965,4 @@ public class ELUtils {
         return result;
     }
 }
+/* @generated */
\ No newline at end of file
